tests/test_mock_hb85a_device.py

```diff
@@ -0,0 +1,105 @@
+# This file is part of ts_ess_common.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import asyncio
+import logging
+import typing
+import unittest
+
+from lsst.ts.ess import common
+
+logging.basicConfig(
+    format="%(asctime)s:%(levelname)s:%(name)s:%(message)s", level=logging.DEBUG
+)
+
+
+class MockDeviceTestCase(unittest.IsolatedAsyncioTestCase):
+    async def _callback(
+        self, reply: typing.Dict[str, typing.List[typing.Union[str, float]]]
+    ) -> None:
+        self.reply: typing.Optional[
+            typing.Dict[str, typing.List[typing.Union[str, float]]]
+        ] = reply
+
+    async def _check_mock_hx85a_device(
+        self, md_props: common.MockDeviceProperties
+    ) -> None:
+        """Check the working of the MockDevice."""
+        self.data = None
+        self.log = logging.getLogger(type(self).__name__)
+        mtt = common.MockTestTools()
+        sensor = common.sensor.Hx85aSensor(log=self.log)
+        mock_device = common.device.MockDevice(
+            name=md_props.name,
+            device_id="MockDevice",
+            sensor=sensor,
+            callback_func=self._callback,
+            log=self.log,
+            missed_channels=md_props.missed_channels,
+            in_error_state=md_props.in_error_state,
+        )
+
+        await mock_device.open()
```

Comment:
I strongly suggest (on this ticket or a later ticket) that you make devices async context managers. This would allow you to write your tests as:
```
async with common.device.MockDevice(...) as mock_device:
    # Use your mock device here. Do NOT open or close it.
```
Users don't have to remember to open or close a device, and if a test fails, the device will still be closed (unlike your current tests, though I realize that probably doesn't really matter for mock devices).

It's a trivial change. Just add these two methods to `BaseDevice`:
```
async def __aenter__(self) -> BaseDevice:
    await self.open()
    return self

async def __aexit__(
    self,
    type: typing.Optional[typing.Type[BaseException]],
    value: typing.Optional[BaseException],
    traceback: typing.Optional[types.TracebackType],
) -> None:
    await self.close()
```

Reply:
This is a really cool suggestion, thanks! I ran into an issue where the type annotation for `__aenter__` couldn't resolve `BaseDevice` which can be solved in two ways: either make the type annotation a string `"-> BaseDevice"` or add `from __future__ import annotations`. I chose the latter but wonder if the former would be acceptable as well?